pkg/plugin/kms_v2_server.go

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft and contributors.  All rights reserved.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Azure/kubernetes-kms/pkg/config"
+	"github.com/Azure/kubernetes-kms/pkg/metrics"
+
+	"github.com/Azure/kubernetes-kms/pkg/version"
+	"k8s.io/klog/v2"
+	kmsv2 "k8s.io/kms/apis/v2"
+)
+
+// KeyManagementServiceV2Server is a gRPC server.
+type KeyManagementServiceV2Server struct {
+	kvClient Client
+	reporter metrics.StatsReporter
+}
+
+// NewV2Server creates an instance of the KMS Service Server with v2 apis.
+func NewV2Server(pluginConfig *Config) (*KeyManagementServiceV2Server, error) {
+	cfg, err := config.GetAzureConfig(pluginConfig.ConfigFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	kvClient, err := newKeyVaultClient(
+		cfg,
+		pluginConfig.KeyVaultName,
+		pluginConfig.KeyName,
+		pluginConfig.KeyVersion,
+		pluginConfig.ProxyMode,
+		pluginConfig.ProxyAddress,
+		pluginConfig.ProxyPort,
+		pluginConfig.ManagedHSM,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &KeyManagementServiceV2Server{
+		kvClient: kvClient,
+		reporter: metrics.NewStatsReporter(),
+	}, nil
+}
+
+// Status returns the health status of the KMS plugin
+func (s *KeyManagementServiceV2Server) Status(_ context.Context, _ *kmsv2.StatusRequest) (*kmsv2.StatusResponse, error) {
+	return &kmsv2.StatusResponse{
+		Version: version.V2APIVersion,
+		Healthz: "ok",
+		KeyId:   s.kvClient.GetAnnotations()["kms.plugin.io.keyid"],
```

Comment:
This is getting a copy of annotations every time for `Status` only to parse out the keyID. Maybe define `Status` as a kv interface that returns the keyID?